src/MainWindow.vala

```diff
@@ -260,28 +258,20 @@ namespace Terminal {
 
             restorable_terminals = new HashTable<string, TerminalWidget> (str_hash, str_equal);
 
-            if (recreate_tabs) {
-                open_tabs ();
+            if (is_first_window) {
```

Comment:
```suggestion
            if (app.get_windows ().length () == 1) {
```

then you can drop both the `window_number` and `is_first_window` properties.

Reply:
That does not work.  The number of windows in the app may change after construction of the window.